code/modules/xenoarchaeology/xenoartifact.dm

```diff
@@ -0,0 +1,491 @@
+	light_color = LIGHT_COLOR_FIRE
+	desc = "A strange alien device. What could it possibly do?"
+	throw_range = 3
+
+	///How much input the artifact is getting from activator traits
+	var/charge = 0
+	///This isn't a requirement anymore. This just affects how effective the charge is 
+	var/charge_req 
+	///Processing type, used for tick
+	var/process_type
+	///List of targted entities for traits
+	var/list/true_target = list() 
+
+	///Associated traits & colour
+	var/material 
+	///activation trait, minor 1, minor 2, minor 3, major, malfunction
+	var/list/traits = list()
+	///Internal list of unallowed traits
+	var/list/blacklist = list()
+	///Touch hint
+	var/datum/xenoartifact_trait/touch_desc
+	///used for special examine circumstance, science goggles & ghosts
+	var/special_desc = "The artifact is made from a"
+	///Description used for label, used because directly adding shit to desc isn't a good idea
+	var/label_desc
+	///How far the artifact can reach
+	var/max_range = 1
+
+	//Used for signaler trait
+	var/code 
+	var/frequency
+	var/datum/radio_frequency/radio_connection
+
+	//Time between uses
+	var/cooldown = 8 SECONDS
+	///Extra time traits can add to the cooldown
+	var/cooldownmod = 0
+	COOLDOWN_DECLARE(xenoa_cooldown)
+
+	///Everytime the artifact is used this increases. When this is successfully proc'd the artifact gains a malfunction and this is lowered.
+	var/malfunction_chance = 0
+	///How much the chance can change in a sinlge itteration
+	var/malfunction_mod = 0.3
+
+	//snowflake variable for shaped
+	var/transfer_prints = FALSE
+
+/obj/item/xenoartifact/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/xenoartifact_pricing)
+	AddComponent(/datum/component/discoverable, XENOA_DP, TRUE) //Same values as original artifacts from exploration
+
+/obj/item/xenoartifact/Initialize(mapload, difficulty)
+	. = ..()
+
+	generate_xenoa_statics() //This wont load if it's already done, aka this wont spam
+
+	material = difficulty //Difficulty is set, in most cases
+	if(!material)
+		material = pickweight(list(XENOA_BLUESPACE = 8, XENOA_PLASMA = 5, XENOA_URANIUM = 3, XENOA_BANANIUM = 1)) //Maint artifacts and similar situations
+
+	var/price
+	var/extra_masks = 0
+	switch(material)
+		if(XENOA_BLUESPACE) //Check xenoartifact_materials.dm for info on artifact materials/types/traits
+			name = "bluespace [name]"
+			generate_traits(GLOB.xenoa_bluespace_blacklist)
+			if(!price)
+				price = pick(100, 200, 300)
+			extra_masks = pick(1)
+
+		if(XENOA_PLASMA)
+			name = "plasma [name]"
+			generate_traits(GLOB.xenoa_plasma_blacklist)
+			if(!price)
+				price = pick(200, 300, 500)
+			malfunction_mod = 0.5
+			extra_masks = pick(1)
+
+		if(XENOA_URANIUM)
+			name = "uranium [name]"
+			generate_traits(GLOB.xenoa_uranium_blacklist, TRUE) 
+			if(!price)
+				price = pick(300, 500, 800) 
+			malfunction_mod = 1
+			extra_masks = pick(1)
+
+		if(XENOA_BANANIUM)
+			name = "bananium [name]"
+			generate_traits()
```

Comment:
Looked into the null reference runtime in the test - this is the case where `generate_traits` is called passing in `null`. From what I can tell, that is a runtime that would consistently happen every time a bananium artifact spawned.

As far as I can tell the solution made to fix this runtime will work, though the comment is currently misleading.